torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -1513,6 +1513,42 @@ class IrParser {
             nullptr);
       }
     }
+    {
+      if (isOptionEnabled(EnableOption::GraphOp)) {
+        auto ptr_op = getOperatorForLiteral(
+            "aten::index_select(Tensor self, int dim, Tensor index) -> Tensor");
+        REGISTER_PARSE_RULE(
+            ptr_op,
+            {
+              MemoryFormat format;
+              std::list<Val*> list_val;
+              std::tie(format, list_val) = getPWFormatValues(
+                  c10::nullopt,
+                  value_map[node->inputs()[0]->unique()],
+                  value_map[node->inputs()[2]->unique()]);
+              auto input = list_val.front();
+              list_val.pop_front();
+              auto dim_value = constant_as<int>(node->input(1));
```

Comment:
if `int dim` is supposed to be constant, it would be nice to add it as a profiled-constant.

e.g. https://github.com/pytorch/pytorch/blob/ee05f47bddfb97b4b292808543d928b3526fc0ca/torch/csrc/jit/codegen/cuda/parser.cpp#L4233-L4251

Reply:
Let's also add tests for this here:
https://github.com/pytorch/pytorch/blob/master/test/test_jit_cuda_fuser.py

something simple as this vvv
https://github.com/pytorch/pytorch/blob/master/test/test_jit_cuda_fuser.py#L4140-L4153

We should also test for 1). `dim` being a constant vs 2). `dim` is a runtime input.

Reply:
I think we also need an entry here for type inference: https://github.com/pytorch/pytorch/blob/ee05f47bddfb97b4b292808543d928b3526fc0ca/torch/csrc/jit/codegen/cuda/type_inference.cpp#L457